docs/workflow-concepts.md

```diff
@@ -1,6 +1,10 @@
 # Core Concepts
 
-This page serves as an introduction into the core concepts of Argo.
+This page serves as an introduction to the core concepts of Argo.
+## The `Server`
+Argo Workflows Server handles the authentication, the UI, and the authentication mechanism (SSO, tokens, or none [server mode]).
+
+
```

Comment:
```suggestion
This page serves as an introduction to the core concepts of Argo.
```

We should revert this section addition now that the link at the bottom of the page exists. 
Reverting this should also fix the remaining `markdownlint` CI issues